diff --git a/src/isomedia/box_funcs.c b/src/isomedia/box_funcs.c
--- a/src/isomedia/box_funcs.c
+++ b/src/isomedia/box_funcs.c
@@ -128,6 +128,7 @@ GF_Err gf_isom_box_parse_ex(GF_Box **outBox, GF_BitStream *bs, u32 parent_type, Bool is_root_box, u64 parent_size)
 		type = otype = gf_bs_read_u32(bs);
 		hdr_size += 4;
 
+#ifndef GPAC_DISABLE_ISOM_FRAGMENTS
 		//check if free as top-level is not a moof
 		if (is_root_box && (type==GF_ISOM_BOX_TYPE_FREE) && !(gf_bs_get_cookie(bs) & GF_ISOM_BS_COOKIE_NO_MABR_PATCH)) {
 			u32 child_type = gf_bs_peek_bits(bs, 32, 4);
@@ -137,6 +138,7 @@ GF_Err gf_isom_box_parse_ex(GF_Box **outBox, GF_BitStream *bs, u32 parent_type, Bool is_root_box, u64 parent_size)
 				type = otype = GF_ISOM_BOX_TYPE_MOOF;
 			}
 		}
+#endif
 
 		/*no size means till end of file - EXCEPT FOR some old QuickTime boxes...*/
 		if (type == GF_ISOM_BOX_TYPE_TOTL)
