--- a/src/isomedia/isom_intern.c
+++ b/src/isomedia/isom_intern.c
@@ -318,6 +318,7 @@
 		GF_CompactSampleGroupBox *csgp = gf_list_get(sampleGroups, i);
 		if (csgp->type != GF_ISOM_BOX_TYPE_CSGP) continue;
 
+#ifndef GPAC_DISABLE_ISOM_FRAGMENTS
 		if (!all_samples) {
 			u32 j=0;
 			for (j=0; j<gf_list_count(child_boxes); j++) {
@@ -326,6 +327,7 @@
 				all_samples += trun->sample_count;
 			}
 		}
+#endif
 		u32 max_samples = all_samples;
 
 		gf_list_rem(sampleGroups, i);
--- a/src/isomedia/drm_sample.c
+++ b/src/isomedia/drm_sample.c
@@ -1429,9 +1429,11 @@
 			if (sinf && sinf->scheme_type) {
 				saiz_aux_info_type = sinf_fmt = sinf->scheme_type->scheme_type;
 			}
+#ifndef GPAC_DISABLE_ISOM_FRAGMENTS
 			//default to cenc for smooth
 			else if (traf->trex->track->moov->mov->is_smooth)
 				saiz_aux_info_type = sinf_fmt = GF_ISOM_CENC_SCHEME;
+#endif
 		}
 		if (!saiz_aux_info_type && (c1==1) && (c2==1)) {
 			GF_LOG(GF_LOG_WARNING, GF_LOG_CONTAINER, ("[iso file] saiz box without flags nor aux info type and no default scheme, ignoring\n"));
@@ -1471,9 +1473,11 @@
 			if (sinf && sinf->scheme_type) {
 				saio_aux_info_type = sinf_fmt = sinf->scheme_type->scheme_type;
 			}
+#ifndef GPAC_DISABLE_ISOM_FRAGMENTS
 			//default to cenc for smooth
 			else if (traf->trex->track->moov->mov->is_smooth)
 				saio_aux_info_type = sinf_fmt = GF_ISOM_CENC_SCHEME;
+#endif
 		}
 		if (!saio_aux_info_type && (c1==1) && (c2==1)) {
 			GF_LOG(GF_LOG_WARNING, GF_LOG_CONTAINER, ("[iso file] saio box without flags nor aux info type and no default scheme, ignoring\n"));
